Sources/VaporToolbox/New.swift

```diff
@@ -0,0 +1,287 @@
+import ArgumentParser
+import Foundation
+import Yams
+
+extension Vapor {
+    struct New: ParsableCommand {
+        static let configuration = CommandConfiguration(abstract: "Generates a new app.")
+
+        @Argument(help: "Name of project and folder.")
+        var name: String
+
+        /// Dynamic variables taken from the template manifest.
+        var variables: [String: Any] = [:]
+
+        /// Options shared by all templates, independent of the manifest.
+        /// They control the build process of the project.
+        struct BuildOptions: ParsableArguments {
+            @Option(
+                name: [.customShort("T"), .long],
+                help: ArgumentHelp("The URL of a Git repository to use as a template.", valueName: "url")
+            )
+            var template: String?
+
+            @Option(help: "Template repository branch to use.")
+            var branch: String?
+
+            @Option(help: ArgumentHelp("The path of the manifest file. Defaults to `manifest.yml`.", valueName: "file"))
+            var manifest: String?
+
+            @Option(name: .shortAndLong, help: ArgumentHelp("The directory to place the new project in.", valueName: "path"))
+            var output: String?
+
+            @Flag(help: "Skips adding a first commit to the newly created repo.")
+            var noCommit: Bool = false
+
+            @Flag(help: "Skips adding a Git repository to the project folder.")
+            var noGit: Bool = false
+
+            @Flag(name: [.customShort("n"), .customLong("no")], help: "Automatically answer no to all questions.")
+            var noQuestions: Bool = false
+
+            @Flag(name: .shortAndLong, help: "Prints additional information.")
+            var verbose: Bool = false
+
+            @Flag(
+                help: ArgumentHelp(
+                    "Dumps template variables as JSON.",
+                    discussion: "This flag is experimental and unstable. Future releases may change its behavior or remove it.",
+                    visibility: .hidden
+                )
+            )
+            var dumpVariables: Bool = false
+        }
+
+        @OptionGroup(title: "Build Options")
+        var buildOptions: BuildOptions
+
+        mutating func run() throws {
+            if self.buildOptions.dumpVariables {
+                let encoder = JSONEncoder()
+                encoder.outputFormatting = .prettyPrinted
+                let jsonData = try encoder.encode(Vapor.manifest?.variables)
+                print(String(data: jsonData, encoding: .utf8)!)
```

Comment:
This could throw if the template contains non-UTF-8 data - we should handle that gracefully